spyder/app/mainwindow.py

```diff
@@ -1434,9 +1434,27 @@ def post_visible_setup(self):
         self.menuBar().raise_()
         self.is_setting_up = False
 
+        # Handle DPI scale changes and show restart message
+        screen = self.window().windowHandle().screen()
+        screen.logicalDotsPerInchChanged.connect(self.show_DPI_change_message)
+
         # Notify that the setup of the mainwindow was finished
         self.sig_setup_finished.emit()
 
+    def show_DPI_change_message(self):
+        """Show message to restart Spyder since the DPI scale changed."""
+        screen = self.window().windowHandle().screen()
+        screen.logicalDotsPerInchChanged.disconnect(
+            self.show_DPI_change_message)
+        QMessageBox.warning(
+            self, _("Warning"),
+            _("A monitor scale change was detected. <br><br>"
+              "For a proper display of Spyder, please be sure that the "
+              "<i>Enable auto high DPI scaling</i> option "

```

Comment:
I think we should automatically change this option to be `True` at this point (i.e. before restarting) and not making a mention to it.

@dalthviz, what happens if users select the custom scaling option and afterwards perform a scale change through the OS options?

Reply:
@ccordoba12 checking the interaction between the custom scaling and the one from the OS, seems like Qt takes the bigger one but, if the bigger one is the OS scale, some widgets like the completion widget don't grow following the scale. Taking this into account, changing automatically the option of HPDI to auto scaling it's actually needed.